src/main/java/appeng/tile/misc/TileCharger.java

```diff
@@ -215,6 +215,57 @@ public TickRateModulation tickingRequest( IGridNode node, int TicksSinceLastCall
 	private boolean doWork()
 	{
 		final ItemStack myItem = this.inv.getStackInSlot( 0 );
+		boolean changed = false;
+
+		if( !myItem.isEmpty() )
+		{
+			final IMaterials materials = AEApi.instance().definitions().materials();
+
+			if( Platform.isChargeable( myItem ) )
+			{
+				final IAEItemPowerStorage ps = (IAEItemPowerStorage) myItem.getItem();
+
+				if( ps.getAEMaxPower( myItem ) > ps.getAECurrentPower( myItem ) )
+				{
+					final double chargeRate = AEApi.instance().registries().charger().getChargeRate( myItem );
+
+					double extractedAmount = this.extractAEPower( chargeRate, Actionable.MODULATE, PowerMultiplier.CONFIG );
+
+					final double missingChargeRate = chargeRate - extractedAmount;
+					final double missingAEPower = ps.getAEMaxPower( myItem ) - ps.getAECurrentPower( myItem );
+					final double toExtract = Math.min( missingChargeRate, missingAEPower );
+
+					try
+					{
+						extractedAmount += this.getProxy().getEnergy().extractAEPower( toExtract, Actionable.MODULATE, PowerMultiplier.ONE );
+					}
+					catch( GridAccessException e1 )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)